### PR TITLE
Add very basic malloc with free

### PR DESCRIPTION
**Pull request by sirkrypt0**
This might come in handy when working with packets. 
Heap size is currently capped at 16MB, which should be enough for the moment.

### CODE REVIEW

**Comment by sirkrypt0** [comment]
I think free shouldn't be too hard, so I'll put that in here later

**Comment by sirkrypt0** [comment]
Test with something like this in `main.c`:
```c
  printf("Allocating 32 byte memory ... \n");
  uint32_t *ptr = malloc(32);
  printf("Is now at: 0x%x\n", ptr);
  printf("Allocating 4 byte memory ... \n");
  uint32_t *ptr2 = malloc(4);
  printf("Is now at: 0x%x\n", ptr2);
  printf("Allocating 4 byte memory ... \n");
  uint32_t *ptr3 = malloc(4);
  printf("Is now at: 0x%x\n", ptr3);
  printf("Freeing ptr\n");
  free(ptr);
  printf("Allocating 32 byte memory ... \n");
  ptr = malloc(32);
  printf("Is now at: 0x%x\n", ptr);
```

**Review by felixauringer — APPROVED** [review]
I've tried it - it seems to work fine. I really appreciate your motivation :+1: I would have mixed up with the alignments and headers about a thousand times...

**Comment by felixauringer** [comment]
Really nice explanation of `malloc`. I love the ASCII memory layout :+1:

File: libc/unistd/sbrk.c [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+#  include <config.h>
+#endif
+
+#include <errno.h>
+#include <stddef.h>
+#include <sys/types.h>
+
+#include "kernel/memory.h"
+
+uint32_t current_heap_address = HEAP_START;
```
Comment (shm0sby) [inline libc/unistd/sbrk.c]: This should be static for reasons. :)

**Review by shm0sby — APPROVED** [review]

**Pull request merged**